OpenRA.Game/Network/Connection.cs

```diff
@@ -87,18 +87,14 @@ protected virtual void Send(byte[] packet)
 
 		protected void AddPacket(ReceivedPacket packet)
 		{
-			lock (receivedPackets)
-				receivedPackets.Add(packet);
+			receivedPackets.Enqueue(packet);
 		}
 
 		public virtual void Receive(Action<int, byte[]> packetFn)
 		{
-			ReceivedPacket[] packets;
-			lock (receivedPackets)
-			{
-				packets = receivedPackets.ToArray();
-				receivedPackets.Clear();
-			}
+			var packets = new List<ReceivedPacket>(receivedPackets.Count);
```

Comment:
Why maintain packets? To avoid `receivedPackets` from being locked?

You can't then initialize `packets` to be a copy of `receivedPackets`?